gap/attr.gi

```diff
@@ -1524,6 +1524,26 @@ function(D)
   return Concatenation(loops, out);
 end);
 
+# Compute all undirected simple circuits by filtering the output
+# of DigraphAllSimpleCircuits
+InstallMethod(DigraphAllUndirectedSimpleCircuits, "for a digraph",
+                     [IsDigraph], function(D)
+    local digraph, cycles, remove, cycle, cycleRev;
+    digraph := DigraphSymmetricClosure(
+        DigraphRemoveAllMultipleEdges(DigraphMutableCopyIfMutable(D)));
+    cycles := Filtered(DigraphAllSimpleCircuits(digraph),
+        c -> Length(c) > 2 or Length(c) = 1);
+    remove := [];
+    for cycle in cycles do
+        if not cycle in remove and Length(cycle) <> 1 then
+            cycleRev := [cycle[1]];
+            Append(cycleRev, Reversed(cycle{[2 .. Length(cycle)]}));
+            Add(remove, cycleRev);
+        fi;
+    od;
```

Comment:
Having `remove` be implemented using [`HashSet`](https://gap-packages.github.io/datastructures/doc/chap8_mj.html#X789A593B7C227BE5) from the datastructures package instead of a list would be more efficient since the function is doing repeated lookups.